iquip/apps/explorer.py

```diff
@@ -30,43 +30,48 @@ def __init__(self, parent: Optional[QWidget] = None):
         layout.addWidget(self.openButton)
 
 
-class FileFinderThread(QThread):
+class _FileFinderThread(QThread):
     """QThread for finding the file list using a command line.
 
     Signals:
-        finished(experimentList, parent): Fetching the file list is finished.
+        fetched(experimentList, widget): The file list is fetched.
 
     Attributes:
-        path: The path of the directory to search experiment files.
-        parent: The widget corresponding to the path.
+        path: The path of the directory to search for experiment files.
+        widget: The widget corresponding to the path.
     """
 
-    finished = pyqtSignal(list, object)
+    fetched = pyqtSignal(list, object)
 
     def __init__(
         self,
         path: str,
-        parent: Union[QTreeWidget, QTreeWidgetItem],
-        callback: Callable[[List[str], Union[QTreeWidget, QTreeWidgetItem]], None]
+        widget: Union[QTreeWidget, QTreeWidgetItem],
+        callback: Callable[[List[str], Union[QTreeWidget, QTreeWidgetItem]], None],
+        parent: Optional[QObject] = None
     ):
-        """
+        """Extended.
+
         Args:
             callback: The callback method called after this thread is finished.
+            See the attributes section in _FileFinderThread.
         """
-        super().__init__()
+        super().__init__(parent=parent)
         self.path = path
-        self.parent = parent
-        self.finished.connect(callback)
+        self.widget = widget
+        self.fetched.connect(callback)
```

Comment:
To be clear, let's give an explicit connection option:
```suggestion
        self.fetched.connect(callback, type=Qt.QueuedConnection)
```

Reply:
It seems to show our purpose well!